app/controllers/posts_controller.rb

```diff
@@ -19,6 +20,14 @@ def create
 		end
 	end
 
+	def publish
+		@post = Post.find(params[:id])
+
+		authorize @post, :edit?
+		@post.published = true
```

Comment:
you do not call `save` here, so the results will not be persisted. Also take a look into `minitest` good time to start adding some test cases!

Reply:
Oh good point, I didn't know if I needed to? I thought it persisted without it cause I remember checking my db. I will investigate further thanks.